packages/api/tests/animal_group.test.js

```diff
@@ -0,0 +1,419 @@
+import animalGroupModel from '../src/models/animalGroupModel.js';
+import animalGroupRelationshipModel from '../src/models/animalGroupRelationshipModel.js';
+import animalBatchGroupRelationshipModel from '../src/models/animalBatchGroupRelationshipModel.js';
+
+describe('Animal Group Tests', () => {
+  let farm;
+  let newOwner;
+
+  function getRequest({ user_id = newOwner.user_id, farm_id = farm.farm_id }, callback) {
+    chai
+      .request(server)
+      .get('/animal_groups')
+      .set('user_id', user_id)
+      .set('farm_id', farm_id)
+      .end(callback);
+  }
+
+  const getRequestAsPromise = util.promisify(getRequest);
+
+  function postRequest(data, { user_id = newOwner.user_id, farm_id = farm.farm_id }, callback) {
+    chai
+      .request(server)
+      .post('/animal_groups')
+      .set('Content-Type', 'application/json')
+      .set('user_id', user_id)
+      .set('farm_id', farm_id)
+      .send(data)
+      .end(callback);
+  }
+
+  const postRequestAsPromise = util.promisify(postRequest);
+
+  function fakeUserFarm(role = 1) {
+    return { ...mocks.fakeUserFarm(), role_id: role };
+  }
+
+  async function returnUserFarms(role) {
+    const [mainFarm] = await mocks.farmFactory();
+    const [user] = await mocks.usersFactory();
+
+    await mocks.userFarmFactory(
+      {
+        promisedUser: [user],
+        promisedFarm: [mainFarm],
+      },
+      fakeUserFarm(role),
+    );
+    return { mainFarm, user };
+  }
+
+  async function makeAnimalGroup(mainFarm, properties) {
+    const [animalGroup] = await mocks.animal_groupFactory({
+      promisedFarm: [mainFarm],
+      properties,
+    });
+    return animalGroup;
+  }
+
+  async function makeAnimal(mainFarm, properties) {
+    const [animal] = await mocks.animalFactory({
+      promisedFarm: [mainFarm],
+      properties,
+    });
+    return animal;
+  }
+
+  async function makeAnimalBatch(mainFarm, properties) {
+    const [animalBatch] = await mocks.animal_batchFactory({
+      promisedFarm: [mainFarm],
+      properties,
+    });
+    return animalBatch;
+  }
+
+  async function makeAnimalGroupRelationship(animalGroup) {
+    const [animalGroupRelationship] = await mocks.animal_group_relationshipFactory({
+      promisedGroup: [animalGroup],
+    });
+    return animalGroupRelationship;
+  }
+
+  async function makeAnimalBatchGroupRelationship(animalGroup) {
+    const [animalBatchGroupRelationship] = await mocks.animal_batch_group_relationshipFactory({
+      promisedGroup: [animalGroup],
+    });
+    return animalBatchGroupRelationship;
+  }
+
+  beforeEach(async () => {
+    [farm] = await mocks.farmFactory();
+    [newOwner] = await mocks.usersFactory();
+  });
+
+  afterAll(async (done) => {
+    await tableCleanup(knex);
+    await knex.destroy();
+    done();
+  });
+
+  // GET TESTS
+  describe('Get animal groups tests', () => {
+    test('All users should get animal groups for their farm', async () => {
+      const roles = [1];
```

Comment:
Why 1 only...?

Reply:
Oops! I was testing this out and wanted it to run faster and forgot to switch it back